kem/kyber/kyber1024/kyber.go

```diff
@@ -273,8 +273,8 @@ func (*scheme) SharedKeySize() int         { return SharedKeySize }
 func (*scheme) CiphertextSize() int        { return CiphertextSize }
 func (*scheme) EncapsulationSeedSize() int { return EncapsulationSeedSize }
 
-func (sk *PrivateKey) Scheme() kem.Scheme { return Scheme }
-func (pk *PublicKey) Scheme() kem.Scheme  { return Scheme }
+func (sk *PrivateKey) Scheme() kem.Scheme { return &scheme{} }
+func (pk *PublicKey) Scheme() kem.Scheme  { return &scheme{} }
```

Comment:
This is still no good as `pk.Scheme() == pk.Scheme()` will fail.

I'd suggest:

```go
var sch kem.Scheme = &scheme{}
func Scheme() kem.Scheme { return sch }
func (sk *PrivateKey) Scheme() kem.Scheme { return sch }
// ...
```

Now the user can't modify the scheme and `pk.Scheme() == pk.Scheme()` still works.

Reply:
Suggestion applied